### ISSUE TITLE
InputEvent triggers immediately after reloading scene

### ISSUE THREAD

**Issue by boruok** [godotengine/godot#71500]
### Godot version

v4.0.beta.custom_build [91fedb60d]

### System information

Ubuntu 22.10 LTS, GTX1060, OpenGL

### Issue description

Looks like **Input.is_action_just_pressed** triggers immediately if you call reload_current_scene which is can't be reproduced in `3.x` branch, dunno if it's expected behiavour for `master` branch

with following scripts, if you try to reload or change current scene it causes cyclic changing current scene.

```gdscript
# preload
extends Node


func _ready() -> void:
	get_tree().change_scene_to_packed(load("res://game.tscn"))
```

```gdscript
# debug.gd
extends Node


#func _process(delta: float) -> void:
func _input(event: InputEvent) -> void:
	if Input.is_action_just_pressed("quit_game"):
		get_tree().quit()
		print("quit game")
	elif Input.is_action_just_pressed("restart_game"):
		get_tree().change_scene_to_packed(load("res://preload.tscn"))
		print("restarted game")
	elif Input.is_action_just_pressed("restart_scene"):
		get_tree().reload_current_scene()
		print("restarted scene")


func _ready() -> void:
	var map := {
		"restart_game" : [ KEY_F1 ],
		"restart_scene" : [ KEY_F2 ],
		"quit_game" : [ KEY_ESCAPE ],
	}

	for k in map:
		for v in map[k]:
			var e := InputEventKey.new()
			if !InputMap.has_action(k):
				InputMap.add_action(k)
			e.keycode = v
			InputMap.action_add_event(k, e)

```

however, if you try capture event within **_process** game reloads correctly

### Steps to reproduce

1. run attachment
2. press F1 to reload game or F2 to reload current scene
3. cyclic reloading!
4. stop
5. uncomment _process and comment _input
6. run attachment

### Minimal reproduction project

[for_3x_branch.zip](https://github.com/godotengine/godot/files/10423871/for_3x_branch.zip)
[for_master_branch.zip](https://github.com/godotengine/godot/files/10423872/for_master_branch.zip)

**Comment by Sauermann** [comment]
Related to #69227

**Comment by Sauermann** [comment]
This issue gets fixed by #69318
I would consider this Issue a duplicate of #69227
Please have a look at the comments of the other issue for a workaround.
